libcnb-package/src/lib.rs

```diff
@@ -31,7 +34,7 @@ pub enum CargoProfile {
     Release,
 }
 
-/// A convenient type alias to use with [`buildpack_package::BuildpackPackage`] or [`BuildpackData`] when you don't required a specialized metadata representation.
+/// A convenient type alias to use when you don't required a specialized metadata representation.
 pub type GenericMetadata = Option<Table>;
```

Comment:
👍 i'd like to see this moved into `libcnb-data`.  maybe with a default generic set too so that if you don't need specialized metadata you don't need to specify it.  e.g.;

```rust
type GenericMetadata = Option<toml::value::Table>;

pub enum BuildpackDescriptor<BM = GenericMetadata> {
    Single(SingleBuildpackDescriptor<BM>),
    Meta(MetaBuildpackDescriptor<BM>),
}

pub struct SingleBuildpackDescriptor<BM = GenericMetadata> { 
  // ...
}

pub struct MetaBuildpackDescriptor<BM = GenericMetadata> { 
  // ...
}

let buildpack_descriptor = toml::from_str::<BuildpackDescriptor>(toml_str)?;
```

Reply:
@colincasey wanna create an issue and/or PR for it so we can continue tracking this outside of this PR? :)

Reply:
https://github.com/heroku/libcnb.rs/issues/663